hudi-common/src/main/java/org/apache/hudi/common/util/queue/BoundedInMemoryExecutor.java

```diff
@@ -168,6 +169,27 @@ public boolean isRemaining() {
   public void shutdownNow() {
     producerExecutorService.shutdownNow();
     consumerExecutorService.shutdownNow();
+    // close queue to force producer stop
+    queue.close();
+  }
+
+  public boolean awaitTermination() {
+    // if current thread has been interrupted before awaitTermination was called, we still give
+    // executor a chance to proceeding. So clear the interrupt flag and reset it if needed before return.
+    boolean interruptedBefore = Thread.interrupted();
```

Comment:
Hi @guanziyue Since we always call `shutdownNow` before `awaitTermination`, here why we need to clear `interrupt` status and await producer&consumer terminated? They should already shutdown by `shutdownNow` before?